crates/matrix-sdk-crypto/src/types/events/utd_cause.rs

```diff
@@ -163,6 +191,47 @@ impl UtdCause {
             _ => UtdCause::Unknown,
         }
     }
+
+    /**
+     * Below is the flow chart we follow for deciding whether historical
+     * UTDs are expected. This function starts at position `B`.
+     *
+     * ```mermaid
+     * flowchart TD
+     * A{Is the message newer than the device?}
+     * A -- No --> B
+     * A -- Yes --> X
+     * B{Is there a backup on the server?}
+     * B -- No --> Y
+     * B -- Yes --> C
+     * C{Is backup working on this device?}
+     * C -- No --> D
+     * C -- Yes --> X
+     * D{Is this device verified?}
+     * D -- No --> Z
+     * D -- Yes --> X
+     *
+     * X([Normal UTD error])
+     * Y([History is not available on this device])
+     * Z([You need to verify this device])
+     * ```
+     */
```

Comment:
Sorry to barge in as an uninvited reviewer, but mermaid is nice if it's going to be rendered, this is a private function inside of the crate and thus will never end up in the rendered docs.

I don't mind the source being here, but can we render this using something like: https://mermaid-ascii.art/ as well?

Reply:
mermaid-ascii produced something terrible, so I just reformatted what we had to be aimed at humans instead of mermaid.

Reply:
68cc73e0d9